src/python/pants/binaries/binary_util.py

```diff
@@ -324,8 +329,8 @@ def __init__(self, baseurls, binary_tool_fetcher, path_by_id=None,
   # to fail until a binary is requested. The HostPlatform should be a parameter that gets lazily
   # resolved by the v2 engine.
   @memoized_method
-  def _host_platform(self):
-    uname_result = self._uname_func()
+  def host_platform(self, uname_func=None):
```

Comment:
Would it be easier to convert this to a memoized class method...? It uses some state from `self`, but it doesn't seem like it really needs to be state on `self` vs on the `cls`.

Failing that: it looks like rather than passing in `uname_func` as an argument, could pass in a uname tuple, and do:
```
uname_result = uname if uname is not None else self._uname_func()
```

Reply:
I had difficulty turning this into a classmethod because `self._path_by_id` is used. However, I implemented your alternative!